src/java.base/share/classes/sun/nio/ch/SelectorImpl.java

```diff
@@ -135,6 +137,24 @@ private int lockAndDoSelect(Consumer<SelectionKey> action, long timeout)
         }
     }
 
+    private int lockAndDoSelect(Consumer<SelectionKey> action, long timeout)
+        throws IOException
+    {
+        // filter selectNow ops from consideration (timeout == 0)
+        if ((timeout == 0) || (!SelectorSelectEvent.enabled())) {
+            return implLockAndDoSelect(action, timeout);
+        }
+        long start = SelectorSelectEvent.timestamp();
+        int n = implLockAndDoSelect(action, timeout);
+        long duration = SelectorSelectEvent.timestamp() - start;
+        // always send event if timed out (n == 0)
+        if ((n == 0) || (SelectorSelectEvent.shouldCommit(duration))) {
```

Comment:
This comment doesn't look right. If n == 0 then it means that no selected keys were added or consumed.